presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -376,6 +376,18 @@ void PrestoServer::run() {
             .sendWithEOM();
       });
 
+  if (systemConfig->prestoNativeSidecar()) {
+    sessionPropertyReporter_ = std::make_unique<SessionPropertyReporter>();
```

Comment:
This memory model is confusing. Seems you have one copy of SessionProperties in SessionPropertyReporter and another in QueryContextManager. There should be only one SessionProperties in PrestoServer. If the correct one is in QueryContextManager because many property translations happen there, then the SessionPropertyReporter should be passed the SessionProperties from the QueryContextManager.

Does this sound reasonable ?